### ISSUE TITLE
suddenly most conda commands fail with "permission denied" error, referencing _multiprocessing.SemLock

### ISSUE THREAD

**Issue by jvollme** [conda/conda#14261]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

We have been working with the current conda installation for quite some time without major problems.
Today, for some reason, we cannot run many conda commands anymore (including ```conda info```,```conda update```, ```conda install`` and many python tools installed under various conda environments... ). They all fail with the same error referencing a "_multiprocessing.SemLock" and a "Permission denied" Error.

Here is the complete error log when trying to run "conda info":
```

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/exception_handler.py", line 17, in __call__
        return func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/cli/main.py", line 83, in main_subshell
        exit_code = do_call(args, parser)
                    ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/cli/conda_argparse.py", line 194, in do_call
        result = getattr(module, func_name)(args, parser)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/cli/main_info.py", line 429, in execute
        info_dict = get_info_dict(args.system)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/cli/main_info.py", line 235, in get_info_dict
        _supplement_index_with_system(virtual_pkg_index)
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/core/index.py", line 177, in _supplement_index_with_system
        for package in context.plugin_manager.get_virtual_packages():
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/manager.py", line 312, in get_virtual_packages
        return tuple(self.get_hook_results("virtual_packages"))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/manager.py", line 189, in get_hook_results
        plugins = [item for items in hook() for item in items]
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/manager.py", line 189, in <listcomp>
        plugins = [item for items in hook() for item in items]
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/virtual_packages/cuda.py", line 65, in conda_virtual_packages
        cuda_version = cached_cuda_version()
                       ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/virtual_packages/cuda.py", line 60, in cached_cuda_version
        return cuda_version()
               ^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/virtual_packages/cuda.py", line 35, in cuda_version
        queue = context.SimpleQueue()
                ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/context.py", line 113, in SimpleQueue
        return SimpleQueue(ctx=self.get_context())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/queues.py", line 341, in __init__
        self._rlock = ctx.Lock()
                      ^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/context.py", line 68, in Lock
        return Lock(ctx=self.get_context())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/synchronize.py", line 167, in __init__
        SemLock.__init__(self, SEMAPHORE, 1, 1, ctx=ctx)
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/synchronize.py", line 57, in __init__
        sl = self._semlock = _multiprocessing.SemLock(
                             ^^^^^^^^^^^^^^^^^^^^^^^^^
    PermissionError: [Errno 13] Permission denied

`$ /opt/conda/miniconda/bin/conda info`

An unexpected error has occurred. Conda has prepared the above report.
If you suspect this error is being caused by a malfunctioning plugin,
consider using the --no-plugins option to turn off plugins.

Example: conda --no-plugins install <package>

Alternatively, you can set the CONDA_NO_PLUGINS environment variable on
the command line to run the command without plugins enabled.

Example: CONDA_NO_PLUGINS=true conda install <package>
```

We have tried running conda with different user permissions, with the NO-PLUGINS option and after running ```conda clean --all``` (which DOES seem to run for some reason). But the problem persists.

What could be the problem here? 

### Conda Info

```shell
# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/exception_handler.py", line 17, in __call__
        return func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/cli/main.py", line 83, in main_subshell
        exit_code = do_call(args, parser)
                    ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/cli/conda_argparse.py", line 194, in do_call
        result = getattr(module, func_name)(args, parser)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/cli/main_info.py", line 429, in execute
        info_dict = get_info_dict(args.system)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/cli/main_info.py", line 235, in get_info_dict
        _supplement_index_with_system(virtual_pkg_index)
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/core/index.py", line 177, in _supplement_index_with_system
        for package in context.plugin_manager.get_virtual_packages():
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/manager.py", line 312, in get_virtual_packages
        return tuple(self.get_hook_results("virtual_packages"))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/manager.py", line 189, in get_hook_results
        plugins = [item for items in hook() for item in items]
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/manager.py", line 189, in <listcomp>
        plugins = [item for items in hook() for item in items]
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/virtual_packages/cuda.py", line 65, in conda_virtual_packages
        cuda_version = cached_cuda_version()
                       ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/virtual_packages/cuda.py", line 60, in cached_cuda_version
        return cuda_version()
               ^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/site-packages/conda/plugins/virtual_packages/cuda.py", line 35, in cuda_version
        queue = context.SimpleQueue()
                ^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/context.py", line 113, in SimpleQueue
        return SimpleQueue(ctx=self.get_context())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/queues.py", line 341, in __init__
        self._rlock = ctx.Lock()
                      ^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/context.py", line 68, in Lock
        return Lock(ctx=self.get_context())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/synchronize.py", line 167, in __init__
        SemLock.__init__(self, SEMAPHORE, 1, 1, ctx=ctx)
      File "/opt/conda/miniconda/lib/python3.11/multiprocessing/synchronize.py", line 57, in __init__
        sl = self._semlock = _multiprocessing.SemLock(
                             ^^^^^^^^^^^^^^^^^^^^^^^^^
    PermissionError: [Errno 13] Permission denied

`$ /opt/conda/miniconda/bin/conda info`


An unexpected error has occurred. Conda has prepared the above report.
If you suspect this error is being caused by a malfunctioning plugin,
consider using the --no-plugins option to turn off plugins.

Example: conda --no-plugins install <package>

Alternatively, you can set the CONDA_NO_PLUGINS environment variable on
the command line to run the command without plugins enabled.

Example: CONDA_NO_PLUGINS=true conda install <package>

If submitted, this report will be used by core maintainers to improve
future releases of conda.
Would you like conda to send this report to the core maintainers? [y/N]: y
Upload successful.

Thank you for helping to improve conda.
Opt-in to always sending reports (and not see this message again)
by running

    $ conda config --set report_errors true
```

### Conda Config

```shell
==> /home/ww5070/.condarc <==
channels:
  - conda-forge
  - bioconda
  - defaults
solver: libmamba

==> envvars <==
no_plugins: True
```

### Conda list

```shell
# packages in environment at /opt/conda/miniconda:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
archspec                  0.2.1              pyhd8ed1ab_1    conda-forge
biopython                 1.81            py311h2582759_0    conda-forge
boltons                   23.0.0             pyhd8ed1ab_0    conda-forge
brotli-python             1.1.0           py311hb755f60_1    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.20.1               hd590300_1    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py311hb3a22ac_0    conda-forge
charset-normalizer        3.3.1              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
conda                     23.11.0         py311h38be061_1    conda-forge
conda-libmamba-solver     23.11.1            pyhd8ed1ab_0    conda-forge
conda-package-handling    2.2.0              pyh38be061_0    conda-forge
conda-package-streaming   0.9.0              pyhd8ed1ab_0    conda-forge
cryptography              41.0.5          py311h63ff55d_0    conda-forge
distro                    1.9.0              pyhd8ed1ab_0    conda-forge
fmt                       10.1.1               h00ab1b0_0    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
jsonpatch                 1.33               pyhd8ed1ab_0    conda-forge
jsonpointer               2.4             py311h38be061_3    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libarchive                3.7.2                h039dbb9_0    conda-forge
libblas                   3.9.0           19_linux64_openblas    conda-forge
libcblas                  3.9.0           19_linux64_openblas    conda-forge
libcurl                   8.4.0                hca28451_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libexpat                  2.5.0                hcb278e6_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_2    conda-forge
libgfortran-ng            13.2.0               h69a702a_2    conda-forge
libgfortran5              13.2.0               ha4646dd_2    conda-forge
libgomp                   13.2.0               h807b86a_2    conda-forge
libiconv                  1.17                 h166bdaf_0    conda-forge
liblapack                 3.9.0           19_linux64_openblas    conda-forge
libmamba                  1.5.3                had39da4_2    conda-forge
libmambapy                1.5.3           py311hf2555c7_2    conda-forge
libnghttp2                1.55.1               h47da74e_0    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libopenblas               0.3.24          pthreads_h413a1c8_0    conda-forge
libsolv                   0.7.25               hfc55251_0    conda-forge
libsqlite                 3.43.2               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_2    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxml2                   2.11.5               h232c23b_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
lz4-c                     1.9.4                hcb278e6_0    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
mamba                     1.5.3           py311h3072747_2    conda-forge
menuinst                  2.0.2           py311h38be061_0    conda-forge
ncurses                   6.4                  hcb278e6_0    conda-forge
numpy                     1.26.0          py311h64a7726_0    conda-forge
openssl                   3.2.1                hd590300_1    conda-forge
packaging                 23.2               pyhd8ed1ab_0    conda-forge
pip                       23.3.1             pyhd8ed1ab_0    conda-forge
platformdirs              4.2.0              pyhd8ed1ab_0    conda-forge
pluggy                    1.3.0              pyhd8ed1ab_0    conda-forge
pybind11-abi              4                    hd8ed1ab_3    conda-forge
pycosat                   0.6.6           py311h459d7ec_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyopenssl                 23.2.0             pyhd8ed1ab_1    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.11.5          hab00c5b_0_cpython    conda-forge
python_abi                3.11                    4_cp311    conda-forge
readline                  8.2                  h8228510_1    conda-forge
reproc                    14.2.4.post0         hd590300_1    conda-forge
reproc-cpp                14.2.4.post0         h59595ed_1    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
ruamel.yaml               0.17.40         py311h459d7ec_0    conda-forge
ruamel.yaml.clib          0.2.7           py311h459d7ec_2    conda-forge
setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h2797004_0    conda-forge
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
truststore                0.8.0              pyhd8ed1ab_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
urllib3                   2.0.7              pyhd8ed1ab_0    conda-forge
wheel                     0.41.2             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml-cpp                  0.8.0                h59595ed_0    conda-forge
zstandard                 0.21.0          py311haa97af0_1    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge
```

**Comment by jezdez** [comment]
@dholth have you seen this before?

**Comment by jvollme** [comment]
yes, and yes

On 23 Sep 2024 15:45, Daniel Holth ***@***.***> wrote:

Are you on Linux and do you have permissions on /dev/shm?

**Comment by jvollme** [comment]
UPDATE: there used to be full permissions for everyone. But apparently this had been changed to full permissions only for root while all other users only have read and excecute, but no write permissions.
I manually changed the permissions back to full permissions for everyone, and now conda is working again.

The question that bugs me now though is: HOW did the permissions for /dev/shm and /dev/mqueue get changed? and WHAT or WHO did that? apparently the owner was still "root"...
